internal/app/tfsec/rules/aws004.go

```diff
@@ -60,49 +58,57 @@ func init() {
 		RequiredLabels:  []string{"aws_lb_listener", "aws_alb_listener"},
 		DefaultSeverity: severity.Error,
 		CheckFunc: func(set result.Set, resourceBlock block.Block, ctx *hclcontext.Context) {
-			if resourceBlock.HasChild("load_balancer_arn") {
-				lbaAttr := resourceBlock.GetAttribute("load_balancer_arn")
-				if lbaAttr.IsResourceBlockReference("aws_lb") {
-					referencedBlock, err := ctx.GetReferencedBlock(lbaAttr)
-					if err == nil {
-						if referencedBlock.HasChild("load_balancer_type") && referencedBlock.GetAttribute("load_balancer_type").Equals("gateway") {
+			// didn't find the referenced block, log and move on
+			if checkIfGateway(resourceBlock, ctx) {
+				return
+			}
+
+			protocolAttr := resourceBlock.GetAttribute("protocol")
+
+			if protocolAttr != nil {
+				if protocolAttr.IsResolvable() && protocolAttr.Equals("HTTPS") {
+					return
+				}
+				if protocolAttr.IsResolvable() && protocolAttr.Equals("HTTP") {
+					// check if this is a redirect to HTTPS - if it is, then no problem
+					if redirectProtocolAttr := resourceBlock.GetAttribute("default_action/redirect/protocol"); redirectProtocolAttr != nil {
```

Comment:
@liamg  - Interested to know your thoughts on this

Reply:
Looks good, I like it :+1: